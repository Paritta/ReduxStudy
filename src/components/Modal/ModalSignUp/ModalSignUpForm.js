import React from "react";
import styled from "styled-components";
import { Field, reduxForm } from "redux-form";
import oc from "open-color";
import PropTypes from "prop-types";

const propTypes = {
    handleSubmit: PropTypes.func,
    hideModal: PropTypes.func,
};

const defaultTypes = {
    handleSubmit() {},
    hideModal() {},
};

const Wrapper = styled.div`
    padding: 30px 0 10px 0;
`;

const StyledField = styled(Field)`
    background-color: ${oc.gray[0]};
    width: 400px;
    height: 40px;
    font-size: 15px;
    border: 2px solid white;
    padding-left: 20px;
    margin-bottom: 10px;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

export let ModalSignUpForm = ({ handleSubmit, hideModal }) => {
    return (
        <Wrapper>
            <form onSubmit={ handleSubmit }>
                <div>
                    <StyledField name="email" component="input" type="text" required placeholder="email"/>
                </div>
                <div>
                    <StyledField name="Nickname" component="input" type="text" required placeholder="Nickname"/>
                </div>
                <div>
                    <StyledField name="password" component="input" type="password" required placeholder="password"/>
                </div>
                <Button type="submit">회원 가입</Button>
                <Button onClick={() => hideModal()}>나가기</Button>
            </form>
        </Wrapper>
    )
};

ModalSignUpForm.propTypes = propTypes;
ModalSignUpForm.defaultTypes = defaultTypes;

ModalSignUpForm = reduxForm({
    form: 'ModalSignUp'
})(ModalSignUpForm);

export default ModalSignUpForm;