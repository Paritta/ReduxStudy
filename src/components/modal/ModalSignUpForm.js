import React from "react";
import styled from "styled-components";

const ModalSignUpForm = () => {
    return (
        <div>
            ModalSignInForm
        </div>
    )
};

export default ModalSignUpForm;