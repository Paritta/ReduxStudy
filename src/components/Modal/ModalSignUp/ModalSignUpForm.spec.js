import React from "react";
import { ModalSignUpForm } from "./ModalSignUpForm";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

it("ModalSignUpForm render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Provider store={store}>
            <ModalSignUpForm />
        </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
