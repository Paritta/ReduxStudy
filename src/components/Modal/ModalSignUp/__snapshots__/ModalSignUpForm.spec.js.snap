// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignUpForm render correctly 1`] = `
.c0 {
  padding: 30px 0 10px 0;
}

.c1 {
  background-color: #f8f9fa;
  width: 400px;
  height: 40px;
  font-size: 15px;
  border: 2px solid white;
  padding-left: 20px;
  margin-bottom: 10px;
}

.c2 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c2:hover {
  background: #f783ac;
  border: 2px solid #f783ac;
}

<div
  className="c0"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        className="c1"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="email"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="c1"
        name="Nickname"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Nickname"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="c1"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="c2"
      type="submit"
    >
      회원 가입
    </button>
    <button
      className="c2"
      onClick={[Function]}
    >
      나가기
    </button>
  </form>
</div>
`;
